api/Port.go

```diff
@@ -151,3 +194,168 @@ func (vrsConnection *VRSConnection) UpdatePortMetadata(name string, metadata map
 
 	return nil
 }
+
+// Disconnected will retry connecting to OVSDB
+// and continue to register for OVSDB updates
+func (ovsdbc OvsdbClient) Disconnected(ovsClient *libovsdb.OvsdbClient) {
+}
+
+// Locked is a placeholder function
+func (ovsdbc OvsdbClient) Locked([]interface{}) {
+}
+
+// Stolen is a placeholder function
+func (ovsdbc OvsdbClient) Stolen([]interface{}) {
+}
+
+// Echo is a placeholder function
+func (ovsdbc OvsdbClient) Echo([]interface{}) {
+}
+
+// Update will provide updates on OVSDB table updates
+func (ovsdbc OvsdbClient) Update(context interface{}, tableUpdates libovsdb.TableUpdates) {
+    ovsdbc.mUpdateChan <- &tableUpdates
+}
+
+// GetNuagePortTableUpdate will register with OVSDB
+// for Nuage Port table updates and return as soon as
+// port table entry gets populated
+func (ovsdbc *OvsdbClient) GetNuagePortTableUpdate() <-chan ClientEvent {
+    var err error
+    var conn *libovsdb.OvsdbClient
+    ovsdbc.mUpdateChan = make(chan *libovsdb.TableUpdates)
+    clientChan := make(chan ClientEvent, 1)
+    for {
+        conn, err = libovsdb.Connect("localhost", 6640)
+        if err != nil {
+            log.Infof("Couldn't connect to ovsdb server while trying to register for Nuage Port table updates")
+        } else {
+            break
+        }
+        time.Sleep(time.Second * time.Duration(5))
+    }
+    log.Infof("Started ovsdb client for monitoring")
+
+    ovsdbc.mConn = conn
+    ovsdbc.mConn.Register(ovsdbc)
+    //set all monitors for ovsdb
+    //set a monitor on Nuage_Port_Table
+    tablesOfInterest := map[string]Empty{"Nuage_Port_Table": {}}
+    monitorRequests := make(map[string]libovsdb.MonitorRequest)
+    schema, ok := ovsdbc.mConn.Schema["Open_vSwitch"]
+    if !ok {
+        log.Infof("Unable to read database schema, ovsdbclient will quit...")
+        return clientChan
+    }
+    for table, tableSchema := range schema.Tables {
+        if _, interesting := tablesOfInterest[table]; interesting {
+            var columns []string
+            for column := range tableSchema.Columns {
+                columns = append(columns, column)
+            }
+            monitorRequests[table] = libovsdb.MonitorRequest{
+                Columns: columns,
+                Select: libovsdb.MonitorSelect{
```

Comment:
Not sure if we need to monitor all of these. Modify must be the only one we need to monitor